### ISSUE TITLE
Issue with Vevo video and some others.

### ISSUE THREAD

**Issue by iancona** [MyToolkit/MyToolkit#35]

**Comment by new01bie** [comment]
same problem, may be youtube change their website source code again

**Comment by Vinquer** [comment]
Same issue here, since the end of November approximately (just to raise the number of persons concerned). 
No error returned but impossible to get a working url on some videos (Vevo mainly)

**Comment by rsuter** [comment]
Can you post some video IDs?

**Comment by Vinquer** [comment]
Sure.
VEVO : https://www.youtube.com/watch?v=oeCihv9A3ac
Not VEVO : https://www.youtube.com/watch?v=6Yl9tgkEbTg

Working: https://www.youtube.com/watch?v=SuBDt7GCbIc

**Comment by mjdavy** [comment]
Looks like YouTube.GetVideoUriAsync is returning an invalid Uri for some videos, including everything I tried from vevo. Check out this playlist for testing https://www.youtube.com/playlist?list=PLtZ7tJkCfjGyPcROkcYAKdeIDXC3aZ9Uw 

Numbers 1,9 and 38 work for me. the others do not. They were all working right before Christmas

**Comment by rajeshkomatlapalli** [comment]
Some Videos from Youtube Viveo are not playing using the MyToolkit.Multimedia.YouTube.GetVideoUriAsync(YoutubeID,MyToolkit.Multimedia.YouTubeQuality.Quality480P);

For Exp: this url worked before 25th Dec "https://www.youtube.com/watch?v=D4sTViBfjVE" ,but the same url is not working today..

Some urls from youtube are working fine, 
 Any modifications in the code to be done, please help.

**Comment by mjdavy** [comment]
A little more info. Since Christmas, I'm getting http status code "Forbidden" on vevo videos using the Uri returned from GetVideoUriAsync. For example this vevo video id "fRh_vgS2dFE" generates a Uri that results in "Forbidden", but  video id "RqQGUJK7Na4" (not vevo) returns success.

**Comment by SmokinLeather** [comment]
I am having this problem too with all the Youtube links that I try. Using Mytoolkit 5.2.8.  Loading the video to a .Net MediaElement.  

This is a link to a video. It has always worked in the past with no problems. Last tested in December.
http://www.youtube.com/watch?v=JT8zLTaKxeE

This is the link returned by Mytoolkit.
https://r9---sn-ab5l6nez.googlevideo.com/videoplayback?signature=A4BB60F1332AC137AA136EEB4F762A74500D3C4C.D980C80E80C2DF9BBC2CFD91D3AA81B6AF90449B&ratebypass=yes&sparams=dur,gcr,id,initcwndbps,ip,ipbits,itag,lmt,mime,mm,mn,ms,mv,pl,ratebypass,requiressl,source,upn,expire&itag=18&key=yt6&pl=31&mime=video/mp4&source=youtube&sver=3&expire=1452543490&upn=OojkOrr4rkU&lmt=1402151308583869&gcr=us&fexp=9408494,9416074,9416126,9417206,9418203,9419452,9420452,9421931,9422517,9422596,9422671,9422781,9423662,9423846,9424388,9425710,9425746&id=o-AP7QFV643Jx_CVAUUqLYXD9ear9fozjDu68jaOBVxPZc&mn=sn-ab5l6nez&mm=31&ipbits=0&initcwndbps=1947500&ip=2601:192:102:8847:80a8:d21d:371:f188&ms=au&mt=1452521792&dur=268.469&mv=m&requiressl=yes

error message from the .Net MediaElement is not very helpful.
Void OpenMedia(System.Uri)... Object reference not set to an instance of an object.... ...    at System.Windows.Media.MediaPlayerState.OpenMedia(Uri source)

PS. Thanks for the Mytoolkit libraries. Awesome work.

**Comment by mjdavy** [comment]
I ended up switching to a modified version of YouTube Extractor. I ported it to work with Universal Windows Apps and added async support. I have not yet had time to submit a pull request to the owner, but meanwhile if you want to go this route and use my mods, you can find them here: https://github.com/mjdavy/TopTastic.net/tree/master/TopTastic in the YouTubeExtractor folder. All videos now work for me.

**Comment by SmokinLeather** [comment]
@mjdavy ,
Thanks for that link. I'll take a look.

**Comment by Englbach** [comment]
I'm having this problem too. Please fix it

**Comment by wisien92** [comment]
Up - same problem

**Comment by rsuter** [comment]
I just used the same YouTubeExtractor code to fix the lib, check out v2.5.10 or directly use YouTubeExtractor. Thx @mjdavy 
Just try again v2.5.10 and report if a video is still not working..

**Comment by SmokinLeather** [comment]
The  v2.5.10 fix did not work for me. Did anyone else try it ? Please let us know if it works for you or not.

**Comment by rsuter** [comment]
What youtube IDs have you tested with?

**Comment by SmokinLeather** [comment]
This on always used to work. But not with  v2.5.10 fix .
http://www.youtube.com/watch?v=JT8zLTaKxeE

**Comment by Englbach** [comment]
Cound not play video on vevo.

**Comment by rsuter** [comment]
@SmokinLeather works for me in Switzerland. In what country are you trying to play the video?

**Comment by SmokinLeather** [comment]
@rsuter , I'm in the United States. Hmm. Maybe it's my app. I'll do some more testing to see what Mtoolkit is returning for a video link. Thanks.

**Comment by rsuter** [comment]
The simplest way is to download the sources and run these unit tests: 

https://github.com/MyToolkit/MyToolkit/blob/master/src/MyToolkit.Tests.WinRT/Multimedia/YouTubeTests.cc

**Comment by SmokinLeather** [comment]
youtube link = http://www.youtube.com/watch?v=JT8zLTaKxeE

This is the link returned from Mytoolkit for this. MediaElement dies when I set it's Source to this Uri. Same error as posted above.

https://r9---sn-ab5l6nle.googlevideo.com/videoplayback?signature=E0254468BA47F01300AB4E6A009E3FA4BD450B26.2FC16E1816F62A226C242AFDF4062B91853210AF&key=yt6&ipbits=0&itag=18&mn=sn-ab5l6nle&mm=31&mime=video/mp4&requiressl=yes&source=youtube&mv=m&mt=1453310875&ms=au&fexp=9407197,9416126,9420452,9421990,9422596,9423662,9423846,9424129,9425745,9425899,9426718&sparams=dur,gcr,id,initcwndbps,ip,ipbits,itag,lmt,mime,mm,mn,ms,mv,pl,ratebypass,requiressl,source,upn,expire&ip=71.234.216.239&lmt=1402151308583869&initcwndbps=1937500&gcr=us&upn=0QXqtmI-GPs&id=o-AD8oznEu1GBDOJnlTeXfCvoq7yNWGPURURvy2P6Rlb7b&ratebypass=yes&pl=18&expire=1453332571&sver=3&dur=268.469

I really have no idea what these link parms are. Is there a reference page that lists the parms and what they mean?

**Comment by wisien92** [comment]
vevo vids working for me in 2.5.10, I'll investigate it further

**Comment by SmokinLeather** [comment]
I am wondering if this is a problem where the .Net MediaElement cannot handle  HTTPS links.

I found this bug report on this issue and the .Net Dev team closed it saying it will not fix it.  
https://connect.microsoft.com/VisualStudio/feedback/details/934355/in-a-wpf-standalone-application-exe-when-the-source-of-a-mediaelement-is-set-to-a-https-uri-it-throws-a-nullreferenceexception

Is there a way to request HTTP (ie not HTTPS) links from MyToolkit? Does YouTube now require HTTPS?

**Comment by SmokinLeather** [comment]
Quick update.

I believe I confirmed that the issue is that the .Net MediaElement cannot handle HTTPS links.

This is the link returned by MyToolkit.
https://r9---sn-ab5l6n76.googlevideo.com/videoplayback?ratebypass=yes&sver=3&lmt=1402151308583869&dur=268.469&source=youtube&gcr=us&mn=sn-ab5l6n76&sparams=dur,gcr,id,initcwndbps,ip,ipbits,itag,lmt,mime,mm,mn,ms,mv,pl,ratebypass,requiressl,source,upn,expire&requiressl=yes&fexp=9416126,9418581,9420452,9422596,9422921,9423581,9423662,9425699,9426055,9427454,9427578&ipbits=0&itag=18&mm=31&ip=71.234.216.239&mv=m&mt=1453330641&ms=au&pl=18&id=o-AHK1DTekgo5w_qzg_tagESleGGJiReeoPR5wP4Ep877F&expire=1453352296&upn=6m9ZH4_lEj0&key=yt6&mime=video/mp4&initcwndbps=556250&signature=1C50030E9EA020C82D3D8D5184020849C2234A5C.8FFF675509057CCFA843E06EA4DC5C8137FD4EAC

The MediaElement throws this error
Exception thrown: 'System.NullReferenceException' in PresentationCore.dll

But the link opened with Windows Media Player, plays fine.

Ugh. MediaElement , what a POS. Thanks  Microsoft. So glad I wrote my app with .Net controls.

**Comment by rajeshkomatlapalli** [comment]
Thanks for the update, It is working for me, for all Video IDs
@Rico Suter

**Comment by Vinquer** [comment]
Working for me too with 2.5.10
Thanks a lot !

**Comment by rsuter** [comment]
Have you tried to just replace https with http?

**Comment by mediaexplorer74** [comment]
> replace https with http?

Hello from 2023 year. I tried v 2.5.16 today.  That (this) bug returned when I tried both https & http cases :( 

Maybe, useragent obsolete... idk.

Update: I experimented https->http at my own code...
Maybe, you mean 
`public static async Task<YouTubeUri[]> GetUrisAsync(string youTubeId, CancellationToken token)`
function in MyToolkit.Extented project ?
I'll try to download your MyToolkit src code....
